src/baseConfig.js

```diff
@@ -0,0 +1,127 @@
+import extend from 'extend';
+
+/**
+ * BaseConfig class
+ */
+export class BaseConfig {
+
+  /**
+   * The defaults of your config
+   * @param {Object}
+   */
+  defaults = {};
+
+  /**
+   * The current values of your config
+   * @type {Object}
+   */
+  current = {};
+
+  /**
+   * Derived classes must set a unique alias
+   * @type {String}
+   */
+  alias
+
+  /**
+   * Creates an instance of BaseConfig. Copies defaults into current
+   * @param  {Object} defaultsOverwrite = {} Optionally extends the defaults
+   */
+  constructor(defaultsOverwrite = {}) {
+    extendIn(this, 'defaults', defaultsOverwrite);
+    this.current = JSON.parse(JSON.stringify(this.defaults));
```

Comment:
By doing a deep clone we are not leveraging defaults at it's fullest. Imagine i would change a value on the this.current object and later delete that property. I would still want the value of the defaults. Using Object.create might be a better fit.

```js
this.current = Object.create(this.defaults);
```